File: qqbot-client/src/main/java/com/mieai/qqbot/client/MediaAsset.java
package com.mieai.qqbot.client;

import com.mieai.qqbot.domain.bot.BotId;
import java.time.Instant;
import java.util.Objects;
import java.util.UUID;

/** Metadata for a bounded local media asset staged for a bot-scoped send. */
public record MediaAsset(UUID id, BotId botId, QqMediaKind kind, String fileName,
        String contentType, long sizeBytes, Instant createdAt) {
    public MediaAsset {
        Objects.requireNonNull(id, "id must not be null");
        Objects.requireNonNull(botId, "botId must not be null");
        Objects.requireNonNull(kind, "kind must not be null");
        if (fileName == null || fileName.isBlank() || fileName.length() > 255) throw new IllegalArgumentException("fileName is invalid");
        if (contentType == null || contentType.isBlank() || contentType.length() > 128) throw new IllegalArgumentException("contentType is invalid");
        if (sizeBytes < 1L) throw new IllegalArgumentException("sizeBytes must be positive");
        Objects.requireNonNull(createdAt, "createdAt must not be null");
    }
}

File: qqbot-client/src/main/kotlin/com/mieai/qqbot/client/MediaAsset.kt
package com.mieai.qqbot.client

import com.mieai.qqbot.domain.bot.BotId
import java.time.Instant
import java.util.UUID

/** Metadata for a bounded local media asset staged for a bot-scoped send. */
data class MediaAsset(
    val id: UUID,
    val botId: BotId,
    val kind: QqMediaKind,
    val fileName: String,
    val contentType: String,
    val sizeBytes: Long,
    val createdAt: Instant,
) {
    init {
        require(!fileName.isBlank() && fileName.length <= 255) { "fileName is invalid" }
        require(!contentType.isBlank() && contentType.length <= 128) { "contentType is invalid" }
        require(sizeBytes >= 1L) { "sizeBytes must be positive" }
    }
}
